### PR TITLE
[tests-only][full-ci]Added function to get expiration date instead of constant expiration date

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
Previously a constant expiration date was used and the nightly was failing since the expiration date is gone date than the current date. This PR adds a function to that gets the expiration date (+10 days with current date).

### Related NIghtly Failed Issue
https://github.com/owncloud/owncloud-sdk/issues/1159

### CODE REVIEW

File: tests/helpers/sharingHelper.js [hunk 1]
```diff
@@ -130,11 +130,23 @@ const getShareIdToken = (resource, resourceType) => {
   return { shareId, shareToken }
 }
 
+/**
+ *
+ *
+ * @returns {String} expiration data (10+ days from current Date)
+ */
+const getExpirationDate = () => {
```
Comment (saw-jan) [inline tests/helpers/sharingHelper.js:138]:
```suggestion
const generateExpirationDate = () => {
```

File: tests/helpers/sharingHelper.js [hunk 2]
```diff
@@ -130,11 +130,23 @@ const getShareIdToken = (resource, resourceType) => {
...
+  const currentDate = new Date()
+  currentDate.setDate(currentDate.getDate() + 10)
+  return currentDate.toISOString().slice(0, 10)
```
Comment (phil-davis) [inline tests/helpers/sharingHelper.js:141]: I suppose the date is something like `2022-10-20` which happens to be a 10-char-long string.
Future readers (and me) are going to wonder which "10" (or "10s") to change if the date offset needs to be different in the future. A comment will help to explain that `slice(0, 10)` is nothing to do with the `+ 10` in the line above.

Comment (SagarGi) [reply tests/helpers/sharingHelper.js:141]: @phil-davis sure thanks for reporting !!

File: tests/sharingTest.js [hunk 3]
```diff
@@ -883,7 +883,7 @@ describe('Main: Currently testing file/folder sharing,', function () {
               ...validAuthHeaders,
               ...applicationFormUrlEncoded
             },
-            body: 'shareType=0&shareWith=' + sharee + '&path=' + fileEncoded + '&expireDate=' + expirationDate
+            body: 'shareType=0&shareWith=' + sharee + '&path=' + fileEncoded + '&expireDate=' + getExpirationDate()
```
Comment (saw-jan) [inline tests/sharingTest.js:886]: call the date generator function somewhere above and use it everywhere below:
```js
const expirationDate = generateExpirationDate()
```
This way we will use same expirationDate for request and response.
```suggestion
            body: 'shareType=0&shareWith=' + sharee + '&path=' + fileEncoded + '&expireDate=' + expirationDate
```

File: tests/helpers/sharingHelper.js [hunk 4]
```diff
@@ -130,11 +130,26 @@ const getShareIdToken = (resource, resourceType) => {
...
+ *
+ *
+ * @returns {String} expiration date (10+ days from current Date)
+ */
+const generateExpirationDate = () => {
+  const currentDate = new Date()
+  const additionalDaysToAdd = 10
+  // adding +10 days to the current data
+  currentDate.setDate(currentDate.getDate() + additionalDaysToAdd)
+  // extracting date in string like '2022-10-10' from currentDate which is in format like '2022-10-10T07:20:33.287Z' with slice
```
Comment (saw-jan) [inline tests/helpers/sharingHelper.js:143]: Also, what if the dates are single digit? Will there be two digits format (e.g.: `01`)
```suggestion
  // extracting date in string like '2022-10-10' from currentDate which is in format like '2022-10-10T07:20:33.287Z'
```

Comment (SagarGi) [reply tests/helpers/sharingHelper.js:143]: yes new Date always give 2 digit format like `20XX-XX-XX`

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Pull request merged**